### PR TITLE
feat: add ui devtools

### PR DESCRIPTION
**Pull request by arjunvegda**
### Description
Add UI DevTools to enhance the development experience with Jotai.

Find up-to-date documentation [here](https://github.com/jotaijs/jotai-devtools/tree/ui-devtools#readme)

**Remaining Tasks**
- [x] Test compatibility with SSR. Works as expected on Next JS with `transpilePackages` enabled - ([https://201egi-3000.preview.csb.app/](https://201egi-3000.preview.csb.app/))
- [x] Gather user feedback
- [x] Add tests
- [x] Handle atoms that return `undefined`
- [x] Add generic error boundary to catch throws
- [x] Add classes to let end users override some styles
- [x] Improve atom list stability. Handle unmounts and unsubscribe gracefully
- [x] Add filter to filter out private atoms
- [x] Update jotai.org docs 

### Demo
- Forkable [CodeSandbox Demo](https://k5p12d.csb.app/)
#### Add it to your project

```sh
# yarn
yarn add jotai-devtools@next

# npm
npm install jotai-devtools@next --save
```

#### Roadmap
_(after the release of the first alpha)_
- Look into Atom graphs
- Look into time travel
- Cosmetic improvements

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   "name": "jotai-devtools",
-  "version": "0.2.0",
+  "version": "0.3.0-next.2",
   "description": "A powerful toolkit to enhance your development experience with Jotai",
   "author": {
     "name": "Arjun Vegda",
     "url": "https://arjunvegda.com"
   },
-  "sideEffects": false,
+  "sideEffects": true,
```
Comment (dai-shi) [inline package.json:9]: curious what requires this?

File: jest.config.ts [hunk 2]
```diff
@@ -37,14 +37,24 @@ const config: Config.InitialOptions = {
   moduleNameMapper: {
     '^jotai-devtools$': '<rootDir>/src/index.ts',
     '^jotai-devtools/(.*)$': '<rootDir>/src/$1.ts',
+    '\\.(css|less|sass|scss)$': '<rootDir>/__mocks__/styleMock.js',
   },
   modulePathIgnorePatterns: ['dist'],
   globals: {
     __DEV__: true,
```
Comment (dai-shi) [inline jest.config.ts:44]: Just to let you know that Jotai repo goes with a new approach. https://github.com/pmndrs/jotai/pull/1742

File: src/utils/types.ts [hunk 3]
```diff
@@ -9,13 +7,3 @@ export type Message = {
   payload?: any;
   state?: any;
 };
-
-export type Options = Parameters<typeof useStore>[0];
```
Comment (dai-shi) [inline src/utils/types.ts:13]: Note that this was more precise. Now, you basically copied it, it doesn't follow when Jotai's `useStore` changes its option.

File: tsconfig.build.json [hunk 4]
```diff
@@ -13,7 +13,6 @@
     "sourceMap": true,
     "forceConsistentCasingInFileNames": true,
     "isolatedModules": true,
-    "exactOptionalPropertyTypes": true,
```
Comment (dai-shi) [inline tsconfig.build.json:16]: Oh, I wish we could keep it. Optionally, `"noUncheckedIndexedAccess": true,` if possible.

**Review by dai-shi — APPROVED** [review]
Looks good to me (I don't read the detail code of DevTools.)

File: tsconfig.build.json (hunk 4 above)
Comment (arjunvegda) [reply tsconfig.build.json:16]: > Oh, I wish we could keep it. 

I was able to make `exactOptionalPropertyTypes`work (it resulted in a better DX, too, thank you!). Fixed it in [this commit](https://github.com/jotaijs/jotai-devtools/pull/30/commits/6c9cbdffa020e9f15e1e0198d18eefae0b2e4f28)

> Optionally, "noUncheckedIndexedAccess": true, if possible.

We haven't changed that, so it should [already be enabled](https://github.com/jotaijs/jotai-devtools/blob/main/tsconfig.build.json#L10) 🙌

File: jest.config.ts (hunk 2 above)
Comment (arjunvegda) [reply jest.config.ts:44]: Nice! I'll revisit this after a few stable releases of `jotai-devtools`

File: tsconfig.build.json (hunk 4 above)
Comment (dai-shi) [reply tsconfig.build.json:16]: oops, my bad > already be enabled.

**Pull request merged**